src/olympia/blocklist/tests/test_cron.py

```diff
@@ -38,342 +38,298 @@
 @override_switch('blocklist_mlbf_submit', active=True)
 class TestUploadToRemoteSettings(TestCase):
     def setUp(self):
-        addon = addon_factory()
-        version_factory(addon=addon)
-        version_factory(addon=addon)
-        self.block = block_factory(
-            addon=addon_factory(
-                version_kw={'version': '1.2b3'},
-                file_kw={'is_signed': True},
-            ),
-            updated_by=user_factory(),
+        self.user = user_factory()
+        self.addon = addon_factory()
+        self.block = block_factory(guid=self.addon.guid, updated_by=self.user)
+
+        self.mocks: dict[str, mock.Mock] = {}
+        for mock_name in (
+            'olympia.blocklist.cron.statsd.incr',
+            'olympia.blocklist.cron.cleanup_old_files.delay',
+            'olympia.blocklist.cron.upload_filter.delay',
```

Comment:
I don't know about the rest but this specifically is a bad idea with the current setup, because `upload_filter` is not currently covered by dedicated tests - it was all covered by this class. So as it stands by mocking it, you are causing the entirety of `upload_filter` task to not be tested at all. Either that mock needs to go or you need to add additional tests for the task.

Reply:
Done